activesupport/test/deprecation_test.rb

```diff
@@ -210,6 +210,21 @@ def test_default_invalid_behavior
     assert_equal ":invalid is not a valid deprecation behavior.", e.message
   end
 
+  def test_custom_behavior
+    custom_behavior_class = Class.new do
+      def call(message, callstack, horizon, gem_name)
+        warn message
+      end
+    end
+    ActiveSupport::Deprecation.behavior = custom_behavior_class.new
+
+    content = capture(:stderr) do
+      ActiveSupport::Deprecation.warn("warning")
+    end
+
+    assert_match(/warning/, content)
```

Comment:
For being very explicit, what do you think about these small tweaks?

```suggestion
    custom_behavior_class = Class.new do
      def call(message, callstack, horizon, gem_name)
        $stderr.puts message
      end
    end
    ActiveSupport::Deprecation.behavior = custom_behavior_class.new

    content = capture(:stderr) do
      ActiveSupport::Deprecation.warn("foo")
    end

    assert_match(/foo/, content)
```